### ISSUE TITLE
Button doesn't communicate danger state to screen reader users

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#8220]
![image](https://user-images.githubusercontent.com/40970507/112925591-b725ac80-90d7-11eb-9220-d082defa6de2.png)

If the danger state is important visual information that calls warning to potentially destructive or irreversible processes/actions we need to communicate that to our screen reader users as well. 

This can be done a few ways, but off the top of my head I might try adding a visually hidden `span` with a unique id that can be referenced in an `aria-describedby` attribute on the button itself something like this perhaps

![image](https://user-images.githubusercontent.com/40970507/112924488-d4f21200-90d5-11eb-9136-ee625e530fa6.png)

**Comment by joshblack** [comment]
@dakahn is there a good rule to cite for this requirement from WCAG?
